sidecar/internal/service/reclaimspace.go

```diff
@@ -88,14 +110,34 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	var deviceSecret map[string]string
+	var volID string
+	if pv.Spec.CSI == nil {
+		volID = pv.Spec.CSI.VolumeHandle
+		if pv.Spec.CSI.NodeStageSecretRef != nil {
+			// Get the secrets from the k8s cluster
+			deviceSecret, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
+			if err != nil {
+				klog.Errorf("Failed to get secret %v", err)
+				return nil, status.Error(codes.InvalidArgument, err.Error())
+			}
+		}
+	}
+
+	//TODO: Configure the stagingPath using cmdline args
+	var stagingPath string = "/var/lib/kubelet/plugins/kubernetes.io/csi/pv/"
 
 	csiReq := &csiReclaimSpace.NodeReclaimSpaceRequest{
-		VolumeId:          "",
+		VolumeId:          volID,
 		VolumePath:        "",
-		StagingTargetPath: "",
-		VolumeCapability:  &csi.VolumeCapability{},
-		Secrets:           map[string]string{},
+		StagingTargetPath: stagingPath + pvName,
```

Comment:
It seems that `StagingTargetPath` in the CSI `NodeStageVolume` contains `/globalmount` as well. It should be included here too.

```
I1223 12:01:43.532455  375675 utils.go:191] ID: 45 Req-ID: 0001-0011-openshift-storage-0000000000000001-96e9a7ce-63c8-11ec-89d0-0a580a83001c GRPC call: /csi.v1.Node/NodeStageVolume
I1223 12:01:43.532652  375675 utils.go:195] ID: 45 Req-ID: 0001-0011-openshift-storage-0000000000000001-96e9a7ce-63c8-11ec-89d0-0a580a83001c GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-720d76c5-d994-4eb8-b8c0-d2bc5f493c17/globalmount
```